common/src/main/java/com/wynntils/features/inventory/CustomBankPagesFeature.java

```diff
@@ -0,0 +1,221 @@
+    public final Config<Integer> buttonFivePage = new Config<>(17);
+
+    @RegisterConfig
+    public final Config<Integer> buttonSixPage = new Config<>(21);
+
+    private static final int MAX_BANK_PAGES = 21;
+    private static final int NEXT_PAGE_SLOT = 8;
+    private static final int PREVIOUS_PAGE_SLOT = 17;
+    private static final List<Integer> BUTTON_SLOTS = List.of(7, 16, 25, 34, 43, 52);
+    private static final List<Integer> QUICK_JUMP_DESTINATIONS = List.of(1, 5, 9, 13, 17, 21);
+    private static final Pattern BANK_PAGE_MATCHER = Pattern.compile("§0\\[Pg\\. (\\d{1,2})\\] §8\\w+'s§0 Bank");
+
+    private boolean quickJumping = false;
+    private int containerId;
+    private int currentPage = 1;
+    private int pageDestination = 1;
+    private List<Integer> customJumpDestinations;
+
+    @SubscribeEvent
+    public void onScreenInit(ScreenInitEvent e) {
+        if (!(e.getScreen() instanceof AbstractContainerScreen<?> screen)) return;
+
+        StyledText title = StyledText.fromComponent(screen.getTitle());
+
+        Matcher pageMatcher = title.getMatcher(BANK_PAGE_MATCHER);
+
+        if (pageMatcher.matches()) {
+            containerId = screen.getMenu().containerId;
+            currentPage = Integer.parseInt(pageMatcher.group(1));
+
+            customJumpDestinations = List.of(
+                    buttonOnePage.get(),
+                    buttonTwoPage.get(),
+                    buttonThreePage.get(),
+                    buttonFourPage.get(),
+                    buttonFivePage.get(),
+                    buttonSixPage.get());
+
+            if (currentPage != pageDestination && quickJumping) {
+                goToPage();
+            } else if (quickJumping) {
+                quickJumping = false;
+            }
+        } else {
+            return;
+        }
+    }
+
+    @SubscribeEvent
+    public void onContainerClose(ContainerCloseEvent.Post e) {
+        currentPage = 1;
+        pageDestination = 1;
+    }
+
+    @SubscribeEvent
+    public void onSlotClicked(ContainerClickEvent e) {
+        if (e.getContainerMenu().containerId != containerId) {
+            return;
+        }
+
+        int slotIndex = e.getSlotNum();
+
+        if (BUTTON_SLOTS.contains(slotIndex)) {
+            int buttonIndex = BUTTON_SLOTS.indexOf(slotIndex);
+            pageDestination = Math.max(1, Math.min(MAX_BANK_PAGES, customJumpDestinations.get(buttonIndex)));
+            e.setCanceled(true);
+            goToPage();
+        } else if (slotIndex == NEXT_PAGE_SLOT) {
+            pageDestination++;
+        } else if (slotIndex == PREVIOUS_PAGE_SLOT) {
+            pageDestination--;
+        }
+    }
+
+    @SubscribeEvent
+    public void onRenderSlot(SlotRenderEvent.Pre e) {
+        if (!(e.getScreen() instanceof AbstractContainerScreen<?> screen)) return;
+
+        StyledText title = StyledText.fromComponent(screen.getTitle());
+
+        Matcher pageMatcher = title.getMatcher(BANK_PAGE_MATCHER);
```

Comment:
Running this regex on every render is expensive, maybe there is a way to check when the page updates instead?

Reply:
Yeah, the render method does way to much. I think we can solve this even without a render method. Can't you just replace the items in the corresponding slots with something that has the wanted icon as texture, and just use vanilla rendering?

Reply:
Had a go at this, I've got it not running the regex each time but obviously it still calls `renderQuickJumpButton` each time.

But I've not been able to get it to replace the item, I'm not even sure what item uses the texture anyway, it might be NBT data.

If you're able to point me in the right direction I can try again, but no rush if no features are being merged until 2.0.3 is released